python_scripts/repeatable/delete_objects.py

```diff
@@ -31,38 +28,59 @@ def parseArguments():
-    parser.add_argument("objectType", help="resources/archival_objects/digital_objects", type=str)
     parser.add_argument("-dR", "--dry-run", help="dry run?", action='store_true')
     parser.add_argument("--version", action="version", version='%(prog)s - Version 1.0')
 
     return parser.parse_args()
 
+def retrieve_object_json(object_uri, local_aspace):
+    """
+    Splits the given URI for the object via / and retrieves the object's JSON.
+    Args:
+        object_uri (DictReader object): the URI of the object
+        local_aspace (ASpaceAPI object): an instance of the ASpace API for connecting to the client
+
+    Returns:
+        object_json (dict): the JSON metadata for the object URI
+    """
+    try:
+        uri_parts = object_uri['uri'].split('/')
+        object_type, object_id = uri_parts[-2], uri_parts[-1]
+    except IndexError as spliterror:
+        record_error(f'retrieve_object_json() - Failed to split URI', spliterror)
+        return None
+    else:
+        if "repositories" in uri_parts:
+            repo_uri = f'repositories/{uri_parts[2]}'
+            object_json = local_aspace.get_object(object_type, object_id, repo_uri)
+        else:
+            object_json = local_aspace.get_object(object_type, object_id)
+        return object_json
+
 
-def main(csv_path, jsonl_path, objectType, dry_run=False):
+def main(csv_path, jsonl_path, dry_run=False):
     """
-    This script takes a CSV of URIs and object type as inputs, grabs all the locations' JSON data using the API, saves
+    This script takes a CSV of URIs and object type as inputs, grabs all the objects' JSON data using the API, saves
     them to a jsonL file using the jsonl_path input, and then deletes them in ArchivesSpace.
 
     The CSV should have the following data structure:
     - Column 1 header = uri
     - Column 1 rows = /locations/#######
 
     Args:
-        csv_path (str): filepath of the CSV file with the location URIs
+        csv_path (str): filepath of the CSV file with the object URIs
         jsonl_path (str): filepath of the jsonL file for storing JSON data of objects before updates - backup
-        objectType (str): the type of object you want to delete. resources/archival_objects/digital_objects/locations
         dry_run (bool): if True, it prints the changed object_json but does not post the changes to ASpace
     """
     local_aspace = ASpaceAPI(os.getenv('as_api'), os.getenv('as_un'), os.getenv('as_pw'))
     for uri in read_csv(csv_path):
-        uri_parts = uri['uri'].split('/')
-        location = local_aspace.get_object(objectType, uri_parts[2])
-        write_to_file(jsonl_path, location)
-        if location:
+        object_json = retrieve_object_json(uri, local_aspace)
+        write_to_file(jsonl_path, object_json)
+        if object_json:
             if dry_run:
-                print(f'Deleted location {uri['uri']}')
+                print(f'Deleted object {uri['uri']}')
```

Comment:
This is a total nit-pick, but I believe in all the other dry runs we've used language that made clear that actions weren't actually taken - so instead of "Deleted object" it would be something like "Object would be deleted" or some such.  I could see myself running this and panicking when I started seeing "Deleted object" and worrying that I might have forgotten to add the dry run.